tesseract_urdf/src/mesh.cpp

```diff
@@ -121,3 +157,5 @@ tinyxml2::XMLElement* tesseract_urdf::writeMesh(const std::shared_ptr<const tess
 
   return xml_element;
```

Comment:
If we save a `ConvexMesh` into the URDF as a `<mesh>` element and then load that mesh from the generated URDF, it will become a `Mesh` even though it is technically convex. Will that cause problems downstream in Tesseract/Trajopt? Does Tesseract/Trajopt only think meshes are convex if they are explicitly `ConvexMesh` types?

If so, I think we will need to add the `tesseract:make_convex` attribute whenever a `ConvexMesh` is encountered, such that it gets converted back into a `ConvexMesh` during the parse of the URDF.

An alternative would be to create a `isConvex` utility that is used in the parse to determine if a mesh is convex. Then we could set its type appropriately. Or better yet, we could do away with the `ConvexMesh` type altogether and add the `isConvex` utility directly to the `Mesh` base class

```suggestion
  // If the mesh is a convex mesh, set the `tesseract:make_convex` attribute true
  if (std::dynamic_pointer_cast<const tesseract_geometry::ConvexMesh>(mesh))
  {
    xml_element->SetAttribute("tesseract:make_convex", true);
  }

  return xml_element;
```

Reply:
Per our discussion today and on a little further investigation, it appears that this problem currently exists in `tesseract` now. For `tesseract` URDF version < 2, all collision meshes automatically get converted to convex meshes. So, if you were to write a URDF from a `tesseract` environment, you would be saving new collision meshes that are convex hulls of the shapes that were used to generate the environment (although they would be saved with the notation that they are convex meshes).

https://github.com/tesseract-robotics/tesseract/blob/9658a37c3231bafc845c1cd280174128e88b0992/tesseract_urdf/src/geometry.cpp#L171-L185

> If so, I think we will need to add the tesseract:make_convex attribute whenever a ConvexMesh is encountered, such that it gets converted back into a ConvexMesh during the parse of the URDF.

Adding the block of code in the previous comment should produce the same behavior that `tesseract` has now